test/io.jl

```diff
@@ -93,3 +93,31 @@
         end
     end
 end
+
+@testset "write" begin
+    @testset "buffer size has no effect on data" begin
+        emd3001 = read("$(@__DIR__)/testdata/emd_3001.map", MRCData)
+        buffer_sizes = [1024, 2048, 4096, 100, 42]
+
+        for buffer_size in buffer_sizes
+            @testset "buffer size: $buffer_size" begin
+                # no preallocation
+                io = IOBuffer(read = true, write = true)
+                mem_non_prealloc = @allocated write(io, emd3001; unit_vsize = buffer_size)
+                closewrite(io)
+                seekstart(io)
+                @test read(io, MRCData) == emd3001
+                close(io)
+
+                # with preallocation
+                io = IOBuffer(read = true, write = true)
+                buffer = Vector{Float32}(undef, buffer_size)
+                mem_prealloc = @allocated write(io, emd3001; buffer)
```

Comment:
```suggestion
                mem_prealloc = @allocated write(io, emd3001; buffer=buffer)
```

Reply:
`mem_prealloc` and `mem_non_prealloc` don't seem to be used. Also, can you add a test for if a buffer of the wrong type is provided?

Reply:
Oh yes, they are a relict of a test I introduced to check if it saves memory to provide a preallocated buffer (which sometimes doesn't seem to be the case, I don't know why...).

And I'll add a test for the data type.